datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -281,7 +280,13 @@ impl<'a> TreeNodeRewriter for ConstEvaluator<'a> {
 
     fn mutate(&mut self, expr: Expr) -> Result<Expr> {
         match self.can_evaluate.pop() {
-            Some(true) => Ok(Expr::Literal(self.evaluate_to_scalar(expr)?)),
+            Some(true) => {
+                let lit = self.evaluate_to_scalar(expr.clone());
+                match lit {
+                    Ok(lit) => Ok(Expr::Literal(lit)),
+                    Err(_) => Ok(expr),
+                }
+            }
```

Comment:
This is the main change

Reply:
This change makes sense to me. Can we please add some rationale / explanation here about *why* the error is ignored so future people who read this code understand 

Perhaps something like
```rust
// Certain expressions such as `CASE` and `COALESCE` are short circuiting
// and may not evalute all their sub expressions. Thus if
// if any error is countered during simplification, return the original 
// so that normal evaluation can occur
```